mimic/test/test_identity_osksadm.py

```diff
@@ -0,0 +1,366 @@
+        self.assertEqual(json_body['conflict']['message'],
+                         "Conflict: Service with the same name already "
+                         "exists.")
+
+    @ddt.data(
+        True, False
+    )
+    def test_successfully_add_service(self, has_id_field):
+        """
+        POST accepts the service type and name regardless of whether
+        an ID field is provided.
+        """
+        data = {
+            'name': self.eeapi.name_key,
+            'type': self.eeapi.type_key,
+            'id': text_type(uuid.uuid4())
+        }
+        if not has_id_field:
+            del data['id']
+
+        req = request(self, self.root, self.verb,
+                      "/identity/v2.0/services",
+                      body=json.dumps(data).encode("utf-8"),
+                      headers=self.headers)
+
+        response = self.successResultOf(req)
+        self.assertEqual(response.code, 201)
+
+    def test_successfully_add_service_with_description(self):
+        """
+        POST accepts a Service Description.
+        """
+        data = {
+            'name': self.eeapi.name_key,
+            'type': self.eeapi.type_key,
+            'description': 'testing external API'
+        }
+        req = request(self, self.root, self.verb,
+                      self.uri,
+                      body=json.dumps(data).encode("utf-8"),
+                      headers=self.headers)
+
+        response = self.successResultOf(req)
+        self.assertEqual(response.code, 201)
+
+
+class TestIdentityMimicOSKSCatalogAdminDeleteExternalService(SynchronousTestCase):
+    """
+    Tests for ``/identity/v2.0/services/<service-id>``, provided by
+    :obj:`mimic.rest.idenity_api.IdentityApi`
+    """
+    def setUp(self):
+        self.core = MimicCore(Clock(), [])
+        self.root = MimicRoot(self.core).app.resource()
+        self.eeapi_id = u"some-id"
+        self.uri = "/identity/v2.0/services/" + self.eeapi_id
+        self.eeapi_name = u"externalServiceName"
+        self.eeapi = make_example_external_api(
+            self,
+            name=self.eeapi_name,
+            set_enabled=True
+        )
+        self.eeapi2 = make_example_external_api(
+            self,
+            name=self.eeapi_name + " alternate"
+        )
+        self.eeapi.uuid_key = self.eeapi_id
+        self.headers = {
+            b'X-Auth-Token': [b'ABCDEF987654321']
+        }
+        self.verb = b"DELETE"
+
+    def test_auth_fail(self):
+        """
+        DELETE with no X-Auth-Token header results in 401.
+        """
+        (response, json_body) = self.successResultOf(
+            json_request(self, self.root, self.verb,
+                         self.uri))
+
+        self.assertEqual(response.code, 401)
+        self.assertEqual(json_body['unauthorized']['code'], 401)
+
+    def test_invalid_service(self):
+        """
+        DELETE an unknown service will generate a 404.
+        """
+        data = {
```

Comment:
There's a lot of code duplication between these tests. Could this be factored out into a common assertion?  It seems like you'd save a lot of lines that way.